dask/dataframe/optimize.py

```diff
@@ -73,14 +81,21 @@ def optimize_read_parquet_getitem(dsk, keys):
                 return dsk
 
             block_columns = block.indices[1][0]
+            if isinstance(block_columns, slice):
+                # only single-column iloc is currently optimized
```

Comment:
What prevents us from doing the same slice on `old.meta.columns[block_columns]` here?

Reply:
Nothing prevents it here, but I noticed that the `set` operation below scrambles the column order with some regularity